packages/api/src/util.js

```diff
@@ -22,3 +22,25 @@ export const shuffle = (arr) => {
     })
     .map((idx) => arr[idx])
 }
+
+
+export const fetchWithTimeout = (url, options) => new Promise((resolve, reject) => {
+  const timeout = setTimeout(() => reject('timeout'), options.timeout || 10 * 1000);
+
+  return fetch(url, options)
+    .then(response => {
+      clearTimeout(timeout);
+
+      return resolve(response)
+      // return a response regardless of the status code.
+      // if (response.status === 200) {
+      //   return resolve(response);
+      // }
+
+      // return reject(response);
+    }, rejectReason => {
+      clearTimeout(timeout);
+
+      return reject(rejectReason);
+    });
+});
```

Comment:
Just added a little check to see if it already timed out:
```suggestion


export const fetchWithTimeout = (url, options) => new Promise((resolve, reject) => {
  let timeout = setTimeout(() => {
    timeout = null;
    reject('timeout');
  }, options.timeout || 10 * 1000);

  return fetch(url, options)
    .then(response => {
      if (timeout === null) {
        // already timed out
        return;
      }
      clearTimeout(timeout);

      return resolve(response)
      // return a response regardless of the status code.
      // if (response.status === 200) {
      //   return resolve(response);
      // }

      // return reject(response);
    }, rejectReason => {
      if (timeout === null) {
        // already timed out
        return;
      }
      clearTimeout(timeout);

      return reject(rejectReason);
    });
});
```